runtime/util/defarg.c

```diff
@@ -25,31 +25,63 @@
 #include "util_internal.h"
 #include "ut_j9util.h"
 
-/* parse arg to determine if it is of the form -Darg=foo, and return foo.
- * Returns an empty string for args of the form -Darg,
- * Returns NULL if the argument is not recognized
+static const char* getDefineArgument(const char *arg, const char *key);
+
+/**
+ * Parse arg to determine if it is of the form -Darg=foo, and return foo.
+ * Returns an empty string for args of the form -Darg.
+ * Returns NULL if the argument is not recognized.
+ *
+ * @param arg the defined argument
+ * @param key the argument key
+ *
+ * @return the argument value or NULL
  */
-char *
-getDefineArgument(char* arg, char* key)
+static const char*
+getDefineArgument(const char *arg, const char *key)
 {
+	const char *result = NULL;
+
 	Trc_Util_getDefineArgument_Entry(arg, key);
-	if (arg[0] == '-' && arg[1] == 'D') {
+	if (('-' == arg[0]) && ('D' == arg[1])) {
 		size_t keyLength = strlen(key);
```

Comment:
Pls pass this as a param instead of computing it every time in the loop.

Reply:
Updated.